From 2132a18096b6a85d79cf3fdf778882fd3b81816b Mon Sep 17 00:00:00 2001
From: Bruno <brunvonlope@outlook.com>
Date: Tue, 4 Jun 2024 10:14:06 -0300
Subject: [PATCH] images/logo: Use reverse DNS naming

---
 images/logo/meson.build | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/images/logo/meson.build b/images/logo/meson.build
index 692bbe4..413d46f 100644
--- a/images/logo/meson.build
+++ b/images/logo/meson.build
@@ -73,12 +73,12 @@ foreach size : desktop_sizes
                            build_by_default: true,
                            install: false)
   meson.add_install_script('sh', '-c',
-                           'mkdir -p "$MESON_INSTALL_DESTDIR_PREFIX/@1@/" && cp -f "@0@" "$MESON_INSTALL_DESTDIR_PREFIX/@1@/gimp.png"'.format(app_icon.full_path(), installdir))
+                           'mkdir -p "$MESON_INSTALL_DESTDIR_PREFIX/@1@/" && cp -f "@0@" "$MESON_INSTALL_DESTDIR_PREFIX/@1@/org.gimp.GIMP.png"'.format(app_icon.full_path(), installdir))
 endforeach
 
 install_data('gimp-logo.svg',
              install_dir: get_option('datadir') / 'icons' / 'hicolor' / 'scalable' / 'apps',
-             rename: 'gimp.svg')
+             rename: 'org.gimp.GIMP.svg')
 
 
 ################################################################################
-- 
2.45.0.windows.1

