From bfd0e0593bcb1ae4735f8178f047c05d58969bb4 Mon Sep 17 00:00:00 2001
From: Luis Diaz Mas <piponazo@gmail.com>
Date: Wed, 16 Sep 2020 10:24:46 +0200
Subject: [PATCH] Use check_cxx_compiler_flags instead of C version

---
 cmake/compilerFlags.cmake | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/cmake/compilerFlags.cmake b/cmake/compilerFlags.cmake
index 63075421..fb9baf38 100644
--- a/cmake/compilerFlags.cmake
+++ b/cmake/compilerFlags.cmake
@@ -1,5 +1,5 @@
 # These flags applies to exiv2lib, the applications, and to the xmp code
-include(CheckCCompilerFlag)
+include(CheckCXXCompilerFlag)
 
 if ( MINGW OR UNIX OR MSYS ) # MINGW, Linux, APPLE, CYGWIN
     if (${CMAKE_CXX_COMPILER_ID} STREQUAL GNU)
@@ -25,9 +25,9 @@ if ( MINGW OR UNIX OR MSYS ) # MINGW, Linux, APPLE, CYGWIN
     if (COMPILER_IS_GCC OR COMPILER_IS_CLANG)
         # This fails under Fedora - MinGW - Gcc 8.3
         if (NOT (MINGW OR CMAKE_HOST_SOLARIS))
-            check_c_compiler_flag(-fstack-clash-protection HAS_FSTACK_CLASH_PROTECTION)
-            check_c_compiler_flag(-fcf-protection HAS_FCF_PROTECTION)
-            check_c_compiler_flag(-fstack-protector-strong HAS_FSTACK_PROTECTOR_STRONG)
+            check_cxx_compiler_flag(-fstack-clash-protection HAS_FSTACK_CLASH_PROTECTION)
+            check_cxx_compiler_flag(-fcf-protection HAS_FCF_PROTECTION)
+            check_cxx_compiler_flag(-fstack-protector-strong HAS_FSTACK_PROTECTOR_STRONG)
             if(HAS_FSTACK_CLASH_PROTECTION)
                 add_compile_options(-fstack-clash-protection)
             endif()
-- 
2.25.4

