From e664124c034599d2e031b6970c218199bcd87dda Mon Sep 17 00:00:00 2001
From: Mathieu Bridon <bochecha@daitauha.fr>
Date: Sun, 24 Apr 2016 13:45:12 +0200
Subject: [PATCH] Don't build/install the documentation

---
 OpenEXR/Makefile.am | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/OpenEXR/Makefile.am b/OpenEXR/Makefile.am
index 3a232af..937133b 100644
--- a/OpenEXR/Makefile.am
+++ b/OpenEXR/Makefile.am
@@ -5,7 +5,7 @@
 ACLOCAL_AMFLAGS = -I m4
 
 SUBDIRS = config IlmImf IlmImfUtil IlmImfTest IlmImfUtilTest \
-	  IlmImfFuzzTest exrheader exrmaketiled IlmImfExamples doc \
+	  IlmImfFuzzTest exrheader exrmaketiled \
 	  exrstdattr exrmakepreview exrenvmap exrmultiview exrmultipart exr2aces
 
 DIST_SUBDIRS = \
