diff -Naur SuiteSparse-5.10.1/Makefile SuiteSparse-5.10.1.patched/Makefile
--- SuiteSparse-5.10.1/Makefile	2021-05-18 15:01:16.000000000 +0200
+++ SuiteSparse-5.10.1.patched/Makefile	2021-07-22 02:03:40.698984882 +0200
@@ -35,7 +35,7 @@
 endif
 	( cd SPQR && $(MAKE) )
 	( cd GraphBLAS && $(MAKE) JOBS=$(JOBS) CMAKE_OPTIONS='$(CMAKE_OPTIONS)' )
-	( cd SLIP_LU && $(MAKE) )
+#	( cd SLIP_LU && $(MAKE) )
 #	( cd PIRO_BAND && $(MAKE) )
 #	( cd SKYLINE_SVD && $(MAKE) )
 
@@ -46,27 +46,27 @@
 # (note that CSparse is not installed; CXSparse is installed instead)
 install: metisinstall
 	( cd SuiteSparse_config && $(MAKE) install )
-	( cd Mongoose  && $(MAKE) CMAKE_OPTIONS='$(CMAKE_OPTIONS)' install )
+#	( cd Mongoose  && $(MAKE) CMAKE_OPTIONS='$(CMAKE_OPTIONS)' install )
 	( cd AMD && $(MAKE) install )
-	( cd BTF && $(MAKE) install )
+#	( cd BTF && $(MAKE) install )
 	( cd CAMD && $(MAKE) install )
 	( cd CCOLAMD && $(MAKE) install )
 	( cd COLAMD && $(MAKE) install )
 	( cd CHOLMOD && $(MAKE) install )
-	( cd CXSparse && $(MAKE) install )
-	( cd LDL && $(MAKE) install )
-	( cd KLU && $(MAKE) install )
+#	( cd CXSparse && $(MAKE) install )
+#	( cd LDL && $(MAKE) install )
+#	( cd KLU && $(MAKE) install )
 	( cd UMFPACK && $(MAKE) install )
-	( cd RBio && $(MAKE) install )
+#	( cd RBio && $(MAKE) install )
 ifneq (,$(GPU_CONFIG))
 	( cd SuiteSparse_GPURuntime && $(MAKE) install )
 	( cd GPUQREngine && $(MAKE) install )
 endif
-	( cd SPQR && $(MAKE) install )
-	( cd GraphBLAS && $(MAKE) JOBS=$(JOBS) CMAKE_OPTIONS='$(CMAKE_OPTIONS)' install )
+#	( cd SPQR && $(MAKE) install )
+#	( cd GraphBLAS && $(MAKE) JOBS=$(JOBS) CMAKE_OPTIONS='$(CMAKE_OPTIONS)' install )
 #	( cd PIRO_BAND && $(MAKE) install )
 #	( cd SKYLINE_SVD && $(MAKE) install )
-	( cd SLIP_LU && $(MAKE) install )
+#	( cd SLIP_LU && $(MAKE) install )
 	$(CP) README.md $(INSTALL_DOC)/SuiteSparse_README.md
 	chmod 644 $(INSTALL_DOC)/SuiteSparse_README.md
 
@@ -110,7 +110,7 @@
 	( cd SuiteSparse_GPURuntime && $(MAKE) uninstall )
 	( cd GPUQREngine && $(MAKE) uninstall )
 	( cd SPQR && $(MAKE) uninstall )
-	( cd SLIP_LU && $(MAKE) uninstall )
+#	( cd SLIP_LU && $(MAKE) uninstall )
 #	( cd PIRO_BAND && $(MAKE) uninstall )
 #	( cd SKYLINE_SVD && $(MAKE) uninstall )
 ifeq (,$(MY_METIS_LIB))
@@ -126,26 +126,26 @@
 # the static library
 library: metis
 	( cd SuiteSparse_config && $(MAKE) )
-	( cd Mongoose  && $(MAKE) CMAKE_OPTIONS='$(CMAKE_OPTIONS)' library )
+#	( cd Mongoose  && $(MAKE) CMAKE_OPTIONS='$(CMAKE_OPTIONS)' library )
 	( cd AMD && $(MAKE) library )
-	( cd BTF && $(MAKE) library )
+#	( cd BTF && $(MAKE) library )
 	( cd CAMD && $(MAKE) library )
 	( cd CCOLAMD && $(MAKE) library )
 	( cd COLAMD && $(MAKE) library )
 	( cd CHOLMOD && $(MAKE) library )
-	( cd KLU && $(MAKE) library )
-	( cd LDL && $(MAKE) library )
+#	( cd KLU && $(MAKE) library )
+#	( cd LDL && $(MAKE) library )
 	( cd UMFPACK && $(MAKE) library )
 	( cd CSparse && $(MAKE) library )
-	( cd CXSparse && $(MAKE) library )
-	( cd RBio && $(MAKE) library )
+#	( cd CXSparse && $(MAKE) library )
+#	( cd RBio && $(MAKE) library )
 ifneq (,$(GPU_CONFIG))
 	( cd SuiteSparse_GPURuntime && $(MAKE) library )
 	( cd GPUQREngine && $(MAKE) library )
 endif
-	( cd SPQR && $(MAKE) library )
-	( cd GraphBLAS && $(MAKE) JOBS=$(JOBS) CMAKE_OPTIONS='$(CMAKE_OPTIONS)' library )
-	( cd SLIP_LU && $(MAKE) library )
+#	( cd SPQR && $(MAKE) library )
+#	( cd GraphBLAS && $(MAKE) JOBS=$(JOBS) CMAKE_OPTIONS='$(CMAKE_OPTIONS)' library )
+#	( cd SLIP_LU && $(MAKE) library )
 #	( cd PIRO_BAND && $(MAKE) library )
 #	( cd SKYLINE_SVD && $(MAKE) library )
 
@@ -173,7 +173,7 @@
 endif
 	( cd SPQR && $(MAKE) static )
 	( cd GraphBLAS && $(MAKE) JOBS=$(JOBS) CMAKE_OPTIONS='$(CMAKE_OPTIONS)' static )
-	( cd SLIP_LU && $(MAKE) static )
+#	( cd SLIP_LU && $(MAKE) static )
 #	( cd PIRO_BAND && $(MAKE) static )
 #	( cd SKYLINE_SVD && $(MAKE) static )
 
@@ -200,7 +200,7 @@
 	- ( cd SuiteSparse_GPURuntime && $(MAKE) purge )
 	- ( cd GPUQREngine && $(MAKE) purge )
 	- ( cd SPQR && $(MAKE) purge )
-	- ( cd SLIP_LU && $(MAKE) purge )
+#	- ( cd SLIP_LU && $(MAKE) purge )
 #	- ( cd PIRO_BAND && $(MAKE) purge )
 #	- ( cd SKYLINE_SVD && $(MAKE) purge )
 	- $(RM) MATLAB_Tools/*/*.mex* MATLAB_Tools/spok/private/*.mex*
@@ -227,7 +227,7 @@
 	- ( cd SuiteSparse_GPURuntime && $(MAKE) clean )
 	- ( cd GPUQREngine && $(MAKE) clean )
 	- ( cd SPQR && $(MAKE) clean )
-	- ( cd SLIP_LU && $(MAKE) clean )
+#	- ( cd SLIP_LU && $(MAKE) clean )
 #	- ( cd PIRO_BAND && $(MAKE) clean )
 #	- ( cd SKYLINE_SVD && $(MAKE) clean )
 
