From 5fcd7fb68be30d7acf320f7e360cd15e7307586d Mon Sep 17 00:00:00 2001
From: Bruno Lopes <brunvonlope@outlook.com>
Date: Thu, 19 Mar 2026 08:05:43 -0300
Subject: [PATCH] app, libgimpwidgets, meson, plug-ins: Patch macOS bundle

---
 app/main.c            | 25 ++++---------------------
 libgimpbase/gimpenv.c | 12 +++++++-----
 2 files changed, 11 insertions(+), 26 deletions(-)

diff --git a/app/main.c b/app/main.c
index 5ae353a200..295d9d5bd3 100644
--- a/app/main.c
+++ b/app/main.c
@@ -353,13 +353,13 @@ gimp_macos_setenv (const char * progname)
       gboolean need_pythonhome = TRUE;
 
       bin_dir = g_path_get_dirname (resolved_path);
-      tmp = g_strdup_printf ("%s/../Resources/lib", bin_dir);
+      tmp = g_strdup_printf ("%s/../Frameworks", bin_dir);
       lib_dir = g_canonicalize_filename (tmp, NULL);
       g_free (tmp);
-      tmp = g_strdup_printf ("%s/../Resources/share", bin_dir);
+      tmp = g_strdup_printf ("%s/../Resources", bin_dir);
       share_dir = g_canonicalize_filename (tmp, NULL);
       g_free (tmp);
-      tmp = g_strdup_printf ("%s/../Resources/etc", bin_dir);
+      tmp = g_strdup_printf ("%s/../SharedSupport", bin_dir);
       etc_dir = g_canonicalize_filename (tmp, NULL);
       g_free (tmp);
 
@@ -390,23 +390,6 @@ gimp_macos_setenv (const char * progname)
             }
         }
 
-      /* Detect we were built in homebrew for MacOS (for PYTHONHOME purposes) */
-      tmp = g_strdup_printf ("%s/../Frameworks/Python.framework", share_dir);
-      if (tmp && !stat (tmp, &sb) && S_ISDIR (sb.st_mode))
-        {
-          g_print ("GIMP was built with homebrew\n");
-          need_pythonhome = FALSE;
-        }
-      g_free (tmp);
-      /* Detect we were built in MacPorts for MacOS (for PYTHONHOME purposes) */
-      tmp = g_strdup_printf ("%s/../Library/Frameworks/Python.framework", share_dir);
-      if (tmp && !stat (tmp, &sb) && S_ISDIR (sb.st_mode))
-        {
-          g_print ("GIMP was built with MacPorts\n");
-          need_pythonhome = FALSE;
-        }
-      g_free (tmp);
-
       /* Minimum runtime paths */
       path_len = strlen (g_getenv ("PATH") ? g_getenv ("PATH") : "") + strlen (bin_dir) + 2;
       path = g_try_malloc (path_len);
@@ -471,7 +454,7 @@ gimp_macos_setenv (const char * progname)
       g_free (tmp);
       if (need_pythonhome)
         {
-          tmp = g_strdup_printf ("%s/Library/Frameworks/Python.framework/Versions/%s", share_dir, PYTHON_VERSION);
+          tmp = g_strdup_printf ("%s/Python.framework/Versions/%s", lib_dir, PYTHON_VERSION);
           g_setenv ("PYTHONHOME", tmp, TRUE);
           g_free (tmp);
         }
diff --git a/libgimpbase/gimpenv.c b/libgimpbase/gimpenv.c
index 8961ad68ba..ba10796164 100644
--- a/libgimpbase/gimpenv.c
+++ b/libgimpbase/gimpenv.c
@@ -434,17 +434,18 @@ gimp_installation_directory (void)
 
   {
     NSAutoreleasePool *pool;
-    NSString          *resource_path;
+    NSString          *app_path;
+    gchar             *app_gpath;
     gchar             *basename;
     gchar             *basepath;
     gchar             *dirname;
 
     pool = [[NSAutoreleasePool alloc] init];
 
-    resource_path = [[NSBundle mainBundle] resourcePath];
+    app_path = [[NSBundle mainBundle] bundlePath];
 
-    basename = g_path_get_basename ([resource_path UTF8String]);
-    basepath = g_path_get_dirname ([resource_path UTF8String]);
+    basename = g_path_get_basename ([app_path UTF8String]);
+    basepath = g_path_get_dirname ([app_path UTF8String]);
     dirname  = g_path_get_basename (basepath);
 
     if (! strcmp (basename, ".libs"))
@@ -507,7 +508,8 @@ gimp_installation_directory (void)
       {
         /*  if none of the above match, we assume that we are really in a bundle  */
 
-        toplevel = g_strdup ([resource_path UTF8String]);
+        app_gpath = g_strdup ([app_path UTF8String]);
+        toplevel = g_strconcat (app_gpath, "/Contents", NULL);
       }
 
     g_free (basename);
-- 
2.50.1 (Apple Git-155)

