From 8bead2a2808447ca4bb00fd2d16bdf51451eff0b Mon Sep 17 00:00:00 2001
From: Bruno Lopes <brunvonlope@outlook.com>
Date: Sat, 7 Feb 2026 18:02:15 -0300
Subject: [PATCH] meson: Patch python version

---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index e6d895cbf8..394d4c9b60 100644
--- a/meson.build
+++ b/meson.build
@@ -130,7 +130,7 @@ versionconfig.set('GIMP_API_VERSION',   gimp_api_version)
 pkgconfig = import('pkgconfig')
 i18n      = import('i18n')
 gnome     = import('gnome')
-python    = import('python').find_installation(modules: 'gi')
+python    = import('python').find_installation('python3.13', modules: 'gi')
 simd      = import('unstable-simd')
 fs        = import('fs')
 
-- 
2.50.1 (Apple Git-155)

