diff -urN gimp-2.10.30-orig/build/windows/gimp-plug-ins.rc.in gimp-2.10.30/build/windows/gimp-plug-ins.rc.in
--- gimp-2.10.30-orig/build/windows/gimp-plug-ins.rc.in	2021-12-20 04:48:33.000000000 +0800
+++ gimp-2.10.30/build/windows/gimp-plug-ins.rc.in	2022-01-05 20:54:00.814712400 +0800
@@ -63,5 +63,5 @@
 END
 
 #include "winuser.h"
-1	ICON	QUOTE(TOP_SRCDIR) "/build/windows/plug-ins.ico"
-CREATEPROCESS_MANIFEST_RESOURCE_ID RT_MANIFEST QUOTE(TOP_SRCDIR) "/build/windows/gimp.manifest"
+1	ICON	"./build/windows/plug-ins.ico"
+CREATEPROCESS_MANIFEST_RESOURCE_ID RT_MANIFEST "./build/windows/gimp.manifest"
diff -urN gimp-2.10.30-orig/build/windows/gimp.rc.in gimp-2.10.30/build/windows/gimp.rc.in
--- gimp-2.10.30-orig/build/windows/gimp.rc.in	2021-12-20 04:48:33.000000000 +0800
+++ gimp-2.10.30/build/windows/gimp.rc.in	2022-01-05 20:52:37.741960900 +0800
@@ -64,6 +64,6 @@
 END
 
 #include "winuser.h"
-1	ICON	QUOTE(TOP_SRCDIR) "/build/windows/wilber.ico"
-2	ICON	QUOTE(TOP_SRCDIR) "/build/windows/fileicon.ico"
-CREATEPROCESS_MANIFEST_RESOURCE_ID RT_MANIFEST QUOTE(TOP_SRCDIR) "/build/windows/gimp.manifest"
+1	ICON	"./build/windows/wilber.ico"
+2	ICON	"./build/windows/fileicon.ico"
+CREATEPROCESS_MANIFEST_RESOURCE_ID RT_MANIFEST "./build/windows/gimp.manifest"
