From 300582409c0ce9e79ba9be0bd346309527b0b79b Mon Sep 17 00:00:00 2001
From: Bruno Lopes <brunvonlope@outlook.com>
Date: Sat, 9 May 2026 20:59:02 -0300
Subject: [PATCH] images: Do not build splash image on MSVC

---
 images/meson.build | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/images/meson.build b/images/meson.build
index 582268d..36a704a 100644
--- a/images/meson.build
+++ b/images/meson.build
@@ -38,10 +38,8 @@ if get_option('windows-installer')
                                 feed: true,
                                 console: true, #needed to not get disordered and lost on fast ninja output and
                                                #to spot very hard-to-debug stuckness (see #11385 and #16202)
-                                build_by_default: true,
+                                build_by_default: false,
                                 env: gimp_run_env)
 endif
 
 subdir('logo')
-
-install_data('sidrat.data', install_dir: gimpdatadir / 'images')
-- 
2.50.1 (Apple Git-155)

