--- a/source/config/mh-mingw	2012-12-28 16:52:21.141964000 -0500
+++ b/source/config/mh-mingw	2012-12-28 18:24:11.955818449 -0500
@@ -66,16 +66,8 @@
 LIBSICU = $(LIBPREFIX)$(STATIC_PREFIX)$(ICUPREFIX)
 A = a
 
-## An import library is needed for z/OS and MSVC
-IMPORT_LIB_EXT = .lib
-
-LIBPREFIX=
-
-# Change the stubnames so that poorly working FAT disks and installation programs can work.
-# This is also for backwards compatibility.
-DATA_STUBNAME = dt
-I18N_STUBNAME = in
-LIBICU = $(STATIC_PREFIX_WHEN_USED)$(ICUPREFIX)
+## import libraries are nice to have under mingw
+IMPORT_LIB_EXT = .dll.a
 
 # The #M# is used to delete lines for icu-config
 # Current full path directory.
--- a/source/config/mh-mingw64	2012-12-28 16:52:21.141964000 -0500
+++ b/source/config/mh-mingw64	2012-12-28 18:24:11.955818449 -0500
@@ -66,16 +66,8 @@
 LIBSICU = $(LIBPREFIX)$(STATIC_PREFIX)$(ICUPREFIX)
 A = a
 
-## An import library is needed for z/OS and MSVC
-IMPORT_LIB_EXT = .lib
-
-LIBPREFIX=
-
-# Change the stubnames so that poorly working FAT disks and installation programs can work.
-# This is also for backwards compatibility.
-DATA_STUBNAME = dt
-I18N_STUBNAME = in
-LIBICU = $(STATIC_PREFIX_WHEN_USED)$(ICUPREFIX)
+## import libraries are nice to have under mingw
+IMPORT_LIB_EXT = .dll.a
 
 # The #M# is used to delete lines for icu-config
 # Current full path directory.
