From 117aeac3bea564a3ed8d237581c8f168e5df889d Mon Sep 17 00:00:00 2001
From: Bruno Lopes <brunvonlope@outlook.com>
Date: Wed, 30 Jul 2025 08:52:34 -0300
Subject: [PATCH] meson: Disable glib-networking test for Snap

---
 meson.build | 52 ----------------------------------------------------
 1 file changed, 52 deletions(-)

diff --git a/meson.build b/meson.build
index dbe316751f..d4e86aae0b 100644
--- a/meson.build
+++ b/meson.build
@@ -488,58 +488,6 @@ rsvg         = dependency('librsvg-2.0', version: '>='+rsvg_minver)
 
 conf.set('PANGO_DISABLE_DEPRECATED',pangocairo.version().version_compare('<1.43'))
 
-
-################################################################################
-# Check for GLib Networking
-
-glib_networking_works_run=false
-if meson.is_cross_build() and not meson.can_run_host_binaries()
-    # Cross-compilation without run capability: we won't be able to
-    # check networking support.
-    glib_networking_works = true
-    glib_warning = '''
-      Test for glib-networking cannot be performed while cross-compiling,
-      unless you set an `exe_wrapper` binary in your toolchain file.
-      Make sure glib-networking is installed, otherwise GIMP will not be able
-      to display the remote help pages through the help browser, nor will it
-      be able to open remote HTTPS (or other protocol using SSL/TLS) files.
-      HTTPS is becoming the expected standard and should not be considered
-      optional anymore.
-    '''
-    warning(glib_warning)
-    warnings += glib_warning
-else # not meson.is_cross_build() or meson.can_run_host_binaries()
-    glib_networking_works_run = cc.run(
-      '''#include <gio/gio.h>
-      int main() {
-        return !g_tls_backend_supports_tls (g_tls_backend_get_default ());
-      }''',
-      dependencies: gio,
-    )
-    glib_networking_works = (glib_networking_works_run.compiled() and
-                             glib_networking_works_run.returncode() == 0)
-    if not glib_networking_works and meson.is_cross_build()
-        # Since cross-platform test runs may be unreliable, let's be
-        # flexible and pass the test with a warning.
-        glib_networking_works = true
-        glib_warning = '''
-      The cross-platform test for glib-networking failed, using the
-      `exe_wrapper` set in your toolchain file.
-      Make sure glib-networking is installed, otherwise GIMP will not be able
-      to display the remote help pages through the help browser, nor will it
-      be able to open remote HTTPS (or other protocol using SSL/TLS) files.
-      HTTPS is becoming the expected standard and should not be considered
-      optional anymore.
-        '''
-        warning(glib_warning)
-        warnings += glib_warning
-    endif
-endif
-
-if not glib_networking_works
-  error('Test for glib-networking failed. This is required.')
-endif
-
 ################################################################################
 # Check if Pango is built with a recent fontconfig
 
-- 
2.49.0.windows.1

